pkg/server/flows_test.go

```diff
@@ -0,0 +1,314 @@
+			"?query={app=\"netobserv-flowcollector\"}|~`SrcPort\":8080`",
+		},
+	}, {
+		inputPath: "?filters=" + url.QueryEscape("SrcPort=8080;SrcAddr=10.128.0.1;SrcK8S_Namespace=default") + "&match=any&reporter=0",
+		outputQueries: []string{
+			"?query={app=\"netobserv-flowcollector\",FlowDirection=\"0\",SrcK8S_Namespace=~\"(?i).*default.*\"}",
+			"?query={app=\"netobserv-flowcollector\",FlowDirection=\"0\"}|json|SrcAddr=ip(\"10.128.0.1\")",
+			"?query={app=\"netobserv-flowcollector\",FlowDirection=\"0\"}|~`SrcPort\":8080`",
+		},
+	}, {
+		inputPath:     "?startTime=1640991600&match=all",
+		outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=1640991600`},
+	}, {
+		inputPath:     "?startTime=1640991600&match=any",
+		outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=1640991600`},
+	}, {
+		inputPath:     "?endTime=1641160800",
+		outputQueries: []string{`?query={app="netobserv-flowcollector"}&end=1641160800`},
+	}, {
+		inputPath:     "?endTime=1641160800&match=any",
+		outputQueries: []string{`?query={app="netobserv-flowcollector"}&end=1641160800`},
+	}, {
+		inputPath:     "?startTime=1640991600&endTime=1641160800&match=all",
+		outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=1640991600&end=1641160800`},
+	}, {
+		inputPath:     "?startTime=1640991600&endTime=1641160800&match=any",
+		outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=1640991600&end=1641160800`},
+		// }, {
+		// 	inputPath:   "?timeRange=300000",
+		// 	outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=${timeNow-300000}`},
+		// }, {
+		// 	inputPath:   "?timeRange=300000&match=any",
+		// 	outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=${timeNow-300000}`},
+		// }, {
+		// 	inputPath:   "?timeRange=86400000&match=all",
+		// 	outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=${timeNow-86400000}`},
+		// }, {
+		// 	inputPath:   "?timeRange=86400000&match=any",
+		// 	outputQueries: []string{`?query={app="netobserv-flowcollector"}&start=${timeNow-86400000}`},
+	}, {
+		inputPath: "?filters=" + url.QueryEscape("SrcK8S_Namespace=\"exact-namespace\""),
+		outputQueries: []string{
+			`?query={app="netobserv-flowcollector",SrcK8S_Namespace="exact-namespace"}`,
+		},
+		// }, {
+		// 	inputPath: "?filters=" + url.QueryEscape("SrcK8S_Namespace=\"start-namespace*\""),
+		// 	outputQueries: []string{
+		// 		`?query={app="netobserv-flowcollector",SrcK8S_Namespace=~"^start-namespace.*"}`,
+		// 	},
+		// }, {
+		// 	inputPath: "?filters=" + url.QueryEscape("SrcK8S_Namespace=\"*end-namespace\""),
+		// 	outputQueries: []string{
+		// 		`?query={app="netobserv-flowcollector",SrcK8S_Namespace=~".*end-namespace$"}`,
+		// 	},
+		// }, {
+		// 	inputPath: "?filters=" + url.QueryEscape("SrcK8S_Namespace=\"mid-n*e\""),
+		// 	outputQueries: []string{
+		// 		`?query={app="netobserv-flowcollector",SrcK8S_Namespace=~"^mid-n.*e$"}`,
+		// 	},
```

Comment:
@jpinsonneau I'd like to get your opinion on that. At the moment, I've removed this mix of exact-matching/start-matching/end-matching, mostly because I don't really see the point of having these particular matches using the double-quotes (which, to me, really mean **exact match**). I guess if you need these start/end matches, you can use the partial match without the double-quotes, no? Or am I missing a use case?

Reply:
sure so as soon as you will find a `*` char in the filter you will go for partial match right ?